.github/workflows/test.yml

```
@@ -0,0 +1,60 @@
+name: Test
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.sol'
+      - '**.yml'
+  push:
+    branches:
+      - main
+    paths:
+      - '**.sol'
+      - '**.yml'
```

I'd remove any filter on paths since there is little to gain from it (faster CI on documentation changes, I suppose).
There could be changes we want to test outside of these files, for example tool configs in json.

```suggestion
  push:
    branches:
      - main
```

Same suggestion for all other *.yml files.

---

I think we should keep this granular. Making CI/CD run faster does have it's benefits, and this would set precedent so that we don't have to consider as much optimising CI/CD at a later time. Additionally, when/if fork tests are included, scoping these become important so as to reduce excessive RPC calls for unrelated PRs. If there are specifically other tools we want to test, I'd also be a fan of them being tested separately (allows CI/CD to run in parallel)

---

I'm a bit worried that some minor change to some other files could break our tooling and we wouldn't notice until there is an unrelated PR. In any case, I'll add a few suggestions to include more extensions.